### PR TITLE
DOCSP-16223 Drop support for zSeries platforms

### PR DESCRIPTION
**Pull request by kanchana-mongodb**
- [JIRA](https://jira.mongodb.org/browse/DOCSP-16223) - requires no doc changes, just updated Release Notes
- [Stage](https://docs-mongodbcom-staging.corp.mongodb.com/42c1da5/bi-connector/docsworker-xlarge/DOCSP-16223/release-notes/#bi-2.14.4)

### CODE REVIEW

File: source/release-notes.txt [hunk 1]
```diff
@@ -20,6 +20,25 @@ Release Notes for |bi|
...
+|bi| 2.14.4
+-----------
+
+*Released <PLEASE UPDATE>, 2021*
+
+Improvements
+~~~~~~~~~~~~
+
+MongoDB now supports only the latest RHEL distribution on IBM zSeries 
+platforms. These zSeries platforms are no longer supported:
```
Comment (rychipman) [inline source/release-notes.txt:34]:
```suggestion
These zSeries platforms are no longer supported:
```

File: source/release-notes.txt [hunk 2]
```diff
@@ -20,6 +20,25 @@ Release Notes for |bi|
...
+
+- ``rhel67-zseries-*``
+- ``suse12-zseries-*``
+- ``suse15-zseries-*``
+- ``ubuntu1604-zseries-*``
+- ``ubuntu1804-zseries-*``
```
Comment (rychipman) [inline source/release-notes.txt:40]:
```suggestion
- ``RHEL 6.7 s390x``
- ``RHEL 7.2 s390x``
- ``SUSE 12 s390x``
- ``SUSE 15 s390x``
- ``Ubuntu 16.04 s390x``
- ``Ubuntu 18.04 s390x``
```

**Review by rychipman — CHANGES_REQUESTED** [review]

**Review by rychipman — APPROVED** [review]
LGTM

File: source/release-notes.txt [hunk 3]
```diff
@@ -20,6 +20,25 @@ Release Notes for |bi|
 .. expect and we don't break the links on the downloads page.
 .. (DOCS-9670 for context)
 
+.. _bi-2-14-4:
+
+|bi| 2.14.4
+-----------
+
+*Released <PLEASE UPDATE>, 2021*
```
Comment (JuliaMongo) [inline source/release-notes.txt:28]: To be safe, I suggest replacing this with May.

**Review by JuliaMongo — APPROVED** [review]
LGTM with one change requested.

**Pull request merged**